js/types.ts

```diff
@@ -1,3 +1,17 @@
+export enum Rock {
```

Comment:
I know that you are having a mystery with importing `Rock` from various places not behaving consistently.  But I think its maybe confusing that it is defined in multiple places like this. 

Maybe you forgot to get rid of this definition while you were experimenting?

Reply:
Right, thanks, it should not be there. But I added it to types.ts when I already had this problem and I was testing if it helps - it doesn't. I've just removed this duplicated declaration in the last commit.

Reply:
Looks good.